tools/cd_scripts/e2e_test.sh

```diff
@@ -40,14 +56,28 @@ set -e
 # Print commands and their arguments as they are executed.
 set -x
 
+# Export the RUN_E2E_TESTS_FOR_ZB_ONLY variable so that it is available in the environment of the 'starterscriptuser' user.
+# Since we are running the subsequent script as 'starterscriptuser' using sudo, the environment of 'starterscriptuser' 
+# would not automatically have access to the environment variables set by the original user (i.e. $RUN_E2E_TESTS_FOR_ZB_ONLY).
+# By exporting this variable, we ensure that the value of RUN_E2E_TESTS_FOR_ZB_ONLY is passed into the 'starterscriptuser' script 
+# and can be used for conditional logic or decisions within that script.
+export RUN_E2E_TESTS_FOR_ZB_ONLY='$RUN_ON_ZB_ONLY'
+
 #Copy details.txt to starterscriptuser home directory and create logs.txt
 cd ~/
 cp /details.txt .
 touch logs.txt
 touch logs-hns.txt
+touch logs-zonal.txt
+GENERIC_LOG_FILE='logs.txt'
+
+if [[ "$RUN_E2E_TESTS_FOR_ZB_ONLY" == "true" ]]; then
+  GENERIC_LOG_FILE='logs-zonal.txt'
+fi
+  
+echo User: $USER &>> ~/${GENERIC_LOG_FILE}
```

Comment:
Put echo parameters in quotes here for better readability.

```suggestion
echo "User: $USER" &>> ~/${GENERIC_LOG_FILE}
```